cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -182,12 +183,20 @@ public void start() {
       LOG.error("Could not disable caching of URLJarFiles. This may lead to 'too many open files` exception.", e);
     }
 
-    ConfigurationLogger.logImportantConfig(cConf);
-
-    LOG.info("Client Hadoop version: {}", ClientVersions.getHadoopVersion());
-    LOG.info("Client ZooKeeper version: {}", ClientVersions.getZooKeeperVersion());
-    LOG.info("Client Kafka version: {}", ClientVersions.getKafkaVersion());
-    LOG.info("Client Hive version: {}", ExploreServiceUtils.getHiveVersion());
+    LOG.info("Hadoop subsystem versions:");
+    LOG.info("  Hadoop version: {}", ClientVersions.getHadoopVersion());
+    LOG.info("  ZooKeeper version: {}", ClientVersions.getZooKeeperVersion());
+    LOG.info("  Kafka version: {}", ClientVersions.getKafkaVersion());
+    if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
+      LOG.info("Client Hive version: {}", ExploreServiceUtils.getHiveVersion());
+    }
+    if (!startupChecker.runSanityChecks()) {
+      LOG.error("Errors while starting up master. " +
+                  "Please check the logs, address all errors, then start the master service again. " +
+                  "Startup checks can also be disabled by setting {} to false in cdap-site.xml",
```

Comment:
Now that I am thinking more - specifying an option to use to disable startup checks might be easy path to disabling it for users. We should remove text for that. What do you guys think ?